components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/dbhandler/StatusDashboardWorkerDBHandler.java

```diff
@@ -79,12 +82,73 @@ public StatusDashboardWorkerDBHandler() {
 
             insertQuery = QueryManager.getInstance().getQuery(SQLConstants.INSERT_QUERY);
             insertQuery = loadQuery(insertQuery, SQLConstants.INSERT_QUERY, dbType);
+
+            // TODO: 11/2/17 proper fix
+            tableCheckQuery = QueryManager.getInstance().getQuery(SQLConstants.ISTABLE_EXISTS_QUERY);
+            tableCheckQuery = loadQuery(tableCheckQuery, SQLConstants.ISTABLE_EXISTS_QUERY, dbType);
+
+            tableCreateQuery = QueryManager.getInstance().getQuery(SQLConstants.CREATE_TABLE);
+            tableCreateQuery = loadQuery(tableCreateQuery, SQLConstants.CREATE_TABLE, dbType);
+
+            creteConfigurationDB();
+            creteDetailsDB();
         } else {
             throw new RDBMSTableException(DATASOURCE_ID + " Could not find. Hence cannot initialize the status " +
                     "dashboard.");
         }
     }
 
+    private void creteConfigurationDB() {
+        Map<String, String> attributesTypeMap = workerAttributeTypeMap.get(WORKER_CONFIG_TABLE);
+        String condition =
+                "WORKERID " + attributesTypeMap.get("WORKERID") + " PRIMARY KEY," +
+                        "HOST " + attributesTypeMap.get("HOST") + "," +
+                        "PORT " + attributesTypeMap.get("PORT");
+        String resolvedTableCreateQuery = tableCreateQuery.replace(PLACEHOLDER_TABLE_NAME, WORKER_CONFIG_TABLE).replace
+                (PLACEHOLDER_COLUMN, condition);
+        Connection conn = this.getConnection();
+        try {
+            PreparedStatement stmt = conn.prepareStatement(resolvedTableCreateQuery);
```

Comment:
What happens when an SQL Exception occurs when this is executed against the DB ?. The statement is not getting closed.